Fix aac_decoder memory leak

QidiWang opened the pull request:
As discussed in [this issue](https://github.com/azahar-emu/azahar/issues/90), some games such as Pokemon X/Y suffer from a memory leak due to NeAACDecInit being called over and over again on each aac frame, when instead it should only be initialized once for each aac stream.

After some testing it seems that a DecoderCommand::Init request is issued whenever a new piece of background music is played, so that should be when we reinitialize the decoder.

QidiWang committed:
Fix aac_decoder memory leak

Only call NeAACDecInit on the first AAC frame and create new NeAACDecoder on DecoderCommand::Init request

QidiWang committed:
update license headers

QidiWang committed:
fix oversight

PabloMK7 commented on src/audio_core/hle/aac_decoder.cpp:
```diff
@@ -52,6 +52,29 @@ BinaryMessage AACDecoder::ProcessRequest(const BinaryMessage& request) {
+        }
+        decoderInitialized = false;
+
+        decoder = NeAACDecOpen();
```
Can you move all this initialization to a AACDecoder::Init function, then we can remove all the code in the AACDecoder constructor. Afterall, there is no need to have the decoder ready until DecoderCommand::Init is received, right?

PabloMK7 commented on src/audio_core/hle/aac_decoder.h:
```diff
@@ -21,6 +21,7 @@ class AACDecoder final : public DecoderBase {
 
     Memory::MemorySystem& memory;
     NeAACDecHandle decoder = nullptr;
+    bool decoderInitialized = false;
```
Please use camel case: decoder_initialized

QidiWang committed:
reorganized code

-put open new FAAD2 AAC decoder code into a separate function
-removed LOG_INFO for open/close FAAD2 AAC decoder
-added LOG_ERROR when no decoder is created to handle decode request, either decoder creation failed or DecoderCommand::Init command not received

QidiWang committed:
Update aac_decoder.cpp

fix clang coding style check

QidiWang committed:
fix load savestate

Loading a savestate creates a situation where decode requests aren't preceded by an init request, so we open a new decoder by default in the constructor. A new AACDecoder instance is always constructed on load savestate.
